apps/s_client.c

```diff
@@ -1311,14 +1313,32 @@ int s_client_main(int argc, char **argv)
         }
     }
     argc = opt_num_rest();
-    if (argc != 0)
+    if (argc == 1) {
+        // If there's a positional argument, it's the equivalent of OP_CONNECT.
+#ifdef AF_UNIX
+        if (socket_family == AF_UNIX) {
```

Comment:
That's an excellent question.

I brought it in from the `OPT_CONNECT` block above. In this case, it seems to be logic that overrides the `OPT_UNIX` flag. Basically it seems that if you pass `-host`, `-port`, or `-connect` in addition to `-unix` then the `-unix` flag is overridden, which makes some sense.

However, I'm definitely retroactively reading intent into the code that's already there. I'm not 100% sure that this code isn't a bit buggy (for example, I haven't checked whether the effect of providing `-connect` and `-unix` depends on the order in which they're applied on the command-line, though it certainly *seems* like it does!).

In essence, then, this code exists to reflect as closely as possible the code in `OPT_CONNECT`.